interfaces/snap_app_set.go

```diff
@@ -196,35 +193,33 @@ func (a *SnapAppSet) SlotRunnables(slot *ConnectedSlot) []snap.Runnable {
 //   - "snap.$snap_instance.{$app1,...$appN, $hook1...$hookN}" if there are
 //     some, but not all, apps/hooks bound
 //   - "snap.$snap_instance.*" if all apps/hooks are bound to the plug or slot
-func labelExpr(apps []*snap.AppInfo, hooks []*snap.HookInfo, appSet *SnapAppSet) string {
+func labelExpr(appSet *SnapAppSet, connected []snap.Runnable) string {
 	var buf bytes.Buffer
 
-	names := make([]string, 0, len(apps)+len(hooks))
-	for _, app := range apps {
-		names = append(names, "."+app.Name)
-	}
+	// all security tags are prefixed with snap.$snap_instance, we use this
+	// knowledge to build a pattern that will match against all of the connected
+	// runnables
+	prefix := fmt.Sprintf("snap.%s", appSet.InstanceName())
 
-	for _, hook := range hooks {
-		if hook.Component != nil {
-			names = append(names, fmt.Sprintf("+%s.hook.%s", hook.Component.Name, hook.Name))
-		} else {
-			names = append(names, fmt.Sprintf(".hook.%s", hook.Name))
-		}
+	suffixes := make([]string, 0, len(connected))
+	for _, r := range connected {
+		suffixes = append(suffixes, strings.TrimPrefix(r.SecurityTag, prefix))
```

Comment:
This `strings.TrimPrefix` is potentially problematic, and we might want to bubble up an error here if the security tag doesn't have the matching prefix. Thoughts, @alfonsosanchezbeato @pedronis?

Reply:
As that would be an internal error maybe we can just log the issue.